lib/segment/src/common/utils.rs

```diff
@@ -5,44 +5,257 @@ use serde_json::Value;
 use crate::data_types::named_vectors::NamedVectors;
 use crate::data_types::vectors::VectorElementType;
 
+/// Avoids allocating Vec with a single element
+pub enum JsonPathTarget<T> {
```

Comment:
It is not used anywhere

Reply:
What do you mean exactly?
The `JsonPathTarget` is used for most functions in this file.